### PR TITLE
create test for url and view and form

### PR DESCRIPTION
**Pull request by mehdiiv**
test view nashod motasefane

### CODE REVIEW

File: building/views.py [hunk 1]
```diff
@@ -12,14 +12,14 @@ class RequiredLoginView(LoginRequiredMixin, TemplateView):
 
 class BuildingCreateView(RequiredLoginView):
     template_name = 'building/form.html'
-    
+
```
Comment (mohsenottello) [inline building/views.py:15]: perfect

File: room/views.py [hunk 2]
```diff
@@ -4,47 +4,68 @@
...
+        self.object = get_object_or_404(Room, id = self.kwargs['room_pk'])
+        return self.object.building_id == self.kwargs['pk']
+
+
+
+class RoomUpdateView(RequiredLoginView):
+    template_name = 'room/form.html'
+
+    def get(self, request , pk, room_pk ):
+        self.object = get_object_or_404(Room, id = self.kwargs['room_pk'])
```
Comment (mohsenottello) [inline room/views.py:53]: please fix two issues:
we dont need `self.object`
we should not use `self.kwargs`, we already have `room_pk`

File: room/views.py [hunk 3]
```diff
@@ -4,47 +4,68 @@
...
+        context = {
+            'form' : RoomForm(instance = self.object)
+        }
+        return render(request, self.template_name, context)
+
+    def post(self, request, pk, room_pk):
+       form = RoomForm(request.POST)
```
Comment (mohsenottello) [inline room/views.py:60]: we are not updating anything, we are creating new record

File: room/views.py (hunk 2 above)
Comment (mohsenottello) [reply room/views.py]: Fixed

File: room/views.py [hunk 4]
```diff
@@ -4,47 +4,68 @@
...
+       form.instance.building_id = pk
+       if form.is_valid():
+           self.object = form.save()
```
[the remark this answers is not in the document]
Comment (mohsenottello) [reply room/views.py]: fixed

File: room/views.py [hunk 5]
```diff
@@ -4,47 +4,68 @@
...
+           return redirect('room_detail', pk, self.object.id)
+       return render(request, 'room/form.html', { 'form': form })
+
+class RoomDeleteView(RequiredLoginView):
+    def get(self, request, pk, room_pk):
+        self.object = get_object_or_404(Room, id = self.kwargs['room_pk'])
```
[the remark this answers is not in the document]
Comment (mohsenottello) [reply room/views.py]: fixed

**Review by mohsenottello — APPROVED** [review]

File: room/views.py (hunk 3 above)
Comment (mehdiiv) [reply room/views.py]: remove

**Pull request merged**